docs/content/python/guides/rbac.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Build Role-Based Access Control (RBAC)
+weight: -1
+description: |
+  Build role-based access control (RBAC) with Oso's built-in authorization
+  modeling features.
+aliases:
+  - todo
+---
+
+# Build Role-Based Access Control (RBAC)
+
+Role-based access control (RBAC) is so ubiquitous that Oso provides special
+syntax for modeling RBAC. This syntax makes it easy to create a role-based
+authorization policy with roles and permissions -- for example, declaring that
+the `"maintainer"` role on a repository allows a user to `"push"` to that
+repository.
+
+In this guide, we'll walk through building an RBAC policy for [GitClub][].
+
+[GitClub]: https://github.com/osohq/gitclub
+
+## Declare application types as actors and resources
+
+Oso makes authorization decisions by determining if an **actor** can perform an
+**action** on a **resource**:
+
+- **Actor**: who is performing the action? `User(id=1)`
+- **Action**: what are they trying to do? `"push"`
+- **Resource**: what are they doing it to? `Repository(id=2)`
+
+The first step of building an RBAC policy is telling Oso which application
+classes are **actors** and which are **resources**. Our example app has a pair
+of **resource** types that we want to control access to, `Organization` and
+`Repository`. We declare both as resources as follows:
+
+```polar
+resource Organization {}
+
+resource Repository {}
+```
+
+Our app also has a `User` class that will be our lone **actor** type:
+
+```polar
+actor User {}
+```
+
+This piece of syntax is called a *resource block*, and it performs two
```

Comment:
Maybe a note in here about registering these classes with `register_class`, could link to another guide for more detail.

Reply:
1b10a4ac